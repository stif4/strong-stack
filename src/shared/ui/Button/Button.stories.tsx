import React from 'react';
import { StoryFn, Meta } from '@storybook/react';
import { Button, ButtonSize, ButtonTheme } from './Button';

export default {
    title: 'shared/Button',
    component: Button,
    argTypes: {
        backgroundColor: { control: 'color' },
        theme: {
            options: [
                ButtonTheme.CLEAR,
                ButtonTheme.OUTLINE,
                ButtonTheme.BACKGROUND,
                ButtonTheme.BACKGROUND_INVERTED,
                'default',
            ],
            control: { type: 'select' },
        },
        size: {
            options: [ButtonSize.M, ButtonSize.L, ButtonSize.XL, 'default'],
            control: { type: 'radio' },
        },
    },
} as Meta<typeof Button>;

const Template: StoryFn<typeof Button> = (args) => <Button {...args} />;

export const Primary = Template.bind({});
Primary.args = {
    children: 'Text',
};

export const Clear = Template.bind({});
Clear.args = {
    children: 'Text',
    theme: ButtonTheme.CLEAR,
};

export const Outline = Template.bind({});
Outline.args = {
    children: 'Text',
    theme: ButtonTheme.OUTLINE,
};

export const Background = Template.bind({});
Background.args = {
    children: 'Text',
    theme: ButtonTheme.BACKGROUND,
};

export const BackgroundInverted = Template.bind({});
BackgroundInverted.args = {
    children: 'Text',
    theme: ButtonTheme.BACKGROUND_INVERTED,
};

export const OutlineSizeM = Template.bind({});
OutlineSizeM.args = {
    children: 'Size M',
    theme: ButtonTheme.OUTLINE,
    size: ButtonSize.M,
};

export const OutlineSizeL = Template.bind({});
OutlineSizeL.args = {
    children: 'Size L',
    theme: ButtonTheme.OUTLINE,
    size: ButtonSize.L,
};

export const OutlineSizeXL = Template.bind({});
OutlineSizeXL.args = {
    children: 'Size XL',
    theme: ButtonTheme.OUTLINE,
    size: ButtonSize.XL,
};

export const SquareSizeXL = Template.bind({});
SquareSizeXL.args = {
    children: '<',
    theme: ButtonTheme.OUTLINE,
    size: ButtonSize.XL,
    square: true,
};

export const AllSizes = () => (
    <div style={{ display: 'flex', gap: '10px', alignItems: 'center' }}>
        {/* eslint-disable-next-line i18next/no-literal-string */}
        <Button size={ButtonSize.M}>Size M</Button>
        {/* eslint-disable-next-line i18next/no-literal-string */}
        <Button size={ButtonSize.L}>Size L</Button>
        {/* eslint-disable-next-line i18next/no-literal-string */}
        <Button size={ButtonSize.XL}>Size XL</Button>
    </div>
);
