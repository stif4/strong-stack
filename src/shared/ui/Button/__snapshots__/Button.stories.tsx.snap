// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shared/Button Clear smoke-test 1`] = `
<div class="app dark">
  <button class="src-shared-ui-Button-Button-module__button src-shared-ui-Button-Button-module__clear">
    Text
  </button>
</div>
`;

exports[`shared/Button Outline smoke-test 1`] = `
<div class="app dark">
  <button class="src-shared-ui-Button-Button-module__button src-shared-ui-Button-Button-module__outline">
    Text
  </button>
</div>
`;

exports[`shared/Button Primary smoke-test 1`] = `
<div class="app dark">
  <button class="src-shared-ui-Button-Button-module__button undefined">
    Text
  </button>
</div>
`;
